x-pack/metricbeat/tests/integration/otel_test.go

```diff
@@ -125,9 +126,16 @@ http.port: {{.MonitoringPort}}
 		},
 		2*time.Minute, 1*time.Second, "Expected at least one ingested metric event, got metricbeat: %d, otel: %d", metricbeatDocs.Hits.Total.Value, otelDocs.Hits.Total.Value)
 
-	otelDoc := otelDocs.Hits.Hits[0]
-	metricbeatDoc := metricbeatDocs.Hits.Hits[0]
-	assertMapstrKeysEqual(t, otelDoc.Source, metricbeatDoc.Source, []string{}, "expected documents keys to be equal")
+	otelDoc := otelDocs.Hits.Hits[0].Source
+	metricbeatDoc := metricbeatDocs.Hits.Hits[0].Source
+	ignoredFields := []string{
+		// only present in beats receivers
+		"otelcol.component.id",
+		"otelcol.component.kind",
+	}
+	assertMapstrKeysEqual(t, otelDoc, metricbeatDoc, ignoredFields, "expected documents keys to be equal")
+	assert.Equal(t, "metricbeatreceiver", otelDoc["otelcol.component.id"], "expected otelcol.component.id field in log record")
+	assert.Equal(t, "receiver", otelDoc["otelcol.component.kind"], "expected otelcol.component.kind field in log record")
```

Comment:
@khushijain21 Hmm, this is a bit weird. The code in libbeat and otelconsumer that sets up these fields is the exact same for filebeat and metricbeat receivers. For some reason, for metricbeat receiver component ID is an empty string. This is fetched from `receiver.Settings.ID.String()`. Any clues why this would return empty string instead of `metricbeatreceiver` for metricbeat?